### ISSUE TITLE
Panic flattening CubicBezierSegment

### ISSUE THREAD

**Issue by dbr** [nical/lyon#652]
With lyon_geom 0.17:

```rust
use lyon_geom::euclid::Point2D;
use lyon_geom::CubicBezierSegment;

fn main() {
    let bez = CubicBezierSegment {
        from: Point2D::new(-1061.0, -3327.0),
        ctrl1: Point2D::new(-1061.0, -3177.0),
        ctrl2: Point2D::new(-1061.0, -3477.0),
        to: Point2D::new(-1061.0, -3327.0),
    };
    for b in bez.flattened(1.0) {
        dbg!(b);
    }
}
```

This outputs:
```
[src/main.rs:12] b = (
    -1061.0000000000002,
    -3287.64139941691,
)
[src/main.rs:12] b = (
    -1061.0,
    -3287.6413994169097,
)
[src/main.rs:12] b = (
    -1060.9999999999998,
    -3311.2565597667635,
)
[src/main.rs:12] b = (
    -1061.0,
    -3342.7434402332356,
)
[src/main.rs:12] b = (
    -1061.0,
    -3366.3586005830903,
)
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', .../lyon_geom-0.17.0/src/flatten_cubic.rs:123:49
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
..which I think is this line in current revision:
https://github.com/nical/lyon/blob/ac3f0bdb77ec228d2650bae4d8907954a28ce3a3/crates/geom/src/flatten_cubic.rs#L117

This was a randomly generated point so isn't necessarily a sane curve, but I wouldn't expect it to panic!

**Comment by nical** [comment]
Thanks for filing this and providing a simple test case. A fix is coming soon.

**Comment by nical** [comment]
Fixed in lyon_geom 0.17.1.

**Comment by dbr** [comment]
Thanks! :partying_face:
